### PR TITLE
add  keyboard controls and fix form

### PR DESCRIPTION
**Pull request by VilppeRiskidev**
added some keyboard controls and fixed some other issues

### CODE REVIEW

File: apps/sertra/components/mod-selector/list.tsx [hunk 1]
```diff
@@ -16,7 +15,14 @@ export const ModRowInfo: React.FC<ModRowInfoProps> = ({
   return (
     <>
       <div className={styles.colSmall}>
-        <Image className={styles.rowIcon} alt="mod icon" src={iconUrl} />
+        {/* eslint-disable-next-line @next/next/no-img-element */}
+        <img
+          height={"32px"}
```
Comment (anttimaki) [inline apps/sertra/components/mod-selector/list.tsx:20]: I prefer `height="32px"` for plain string values. But I guess this is something that the linter should catch on, so feel free to leave it as is. Either way you could ensure that we have a ticket for linting in this project, and that this issue is on the list of things we need a house rule for.

File: apps/sertra/components/mod-selector/search.tsx [hunk 2]
```diff
@@ -18,6 +18,70 @@ export const SearchBox = <T,>({
 }: SearchBoxProps<T>) => {
   const [isFocused, setFocused] = useState<boolean>(false);
 
+  function focusFirstOption(target: EventTarget & HTMLInputElement) {
+    if (
```
Comment (anttimaki) [inline apps/sertra/components/mod-selector/search.tsx:22]: Optional chaining operator can be used to call function too, i.e.
`target.parentElement?.parentElement?.nextElementSibling?.children[0]?.children[0]?.focus();`
would remove the need for the `if` block.

That being said, this approach seems really brittle. Small change in the markup will break the function. [useRef](https://reactjs.org/docs/hooks-reference.html#useref) should probably be used here to tag the first option and the search field.

File: apps/sertra/components/mod-selector/search.tsx [hunk 3]
```diff
@@ -18,6 +18,70 @@ export const SearchBox = <T,>({
...
+      if (e.code === "ArrowDown") {
+        e.preventDefault();
+        if (e.currentTarget) {
+          focusFirstOption(e.currentTarget);
+        }
+      }
+    }
+  };
+
+  const listKeyDownHandler = (
```
Comment (anttimaki) [inline apps/sertra/components/mod-selector/search.tsx:52]: Depending on the amount of keyboard stuff we're going to implement, we might want to consider a library for this to make the code more readable/maintainable. [Mousetrap](https://github.com/ccampbell/mousetrap) has worked for me in the past, but I'm not sure if it's still a relevant option today.

Anyway, that's probably outside the scope of this task. Consider adding a task to ClickUp anyway.

File: apps/sertra/pages/index.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import Head from "next/head";
 import { SWRConfig } from "swr";
 import { getServerListings } from "../api/hooks";
 import { ApiURLs } from "../api/urls";
-import styles from "./ServerList.module.css";
+import styles from "../components/ServerList.module.css";
```
Comment (anttimaki) [inline apps/sertra/pages/index.tsx:6]: This seems unrelated to the PR, is it here on purpose?

**Review by anttimaki — CHANGES_REQUESTED** [review]
Please check if the useRef would improve the solution. Other comments are optional to fix now.

File: apps/sertra/components/mod-selector/list.tsx (hunk 1 above)
Comment (VilppeRiskidev) [reply apps/sertra/components/mod-selector/list.tsx:20]: Those values are actually already set in the styles.rowIcon, so I'll remove them. Those were added because linter wanted to use Image component which would've required width and height props.

File: apps/sertra/pages/index.tsx (hunk 4 above)
Comment (VilppeRiskidev) [reply apps/sertra/pages/index.tsx:6]: 'twas broken, so I fixed it, This could've been a separate pr, but it's so small it seemed reasonable to have it with these changes.

File: apps/sertra/components/mod-selector/search.tsx (hunk 3 above)
Comment (VilppeRiskidev) [reply apps/sertra/components/mod-selector/search.tsx:52]: Added a task for this.

File: apps/sertra/components/mod-selector/list.tsx (hunk 1 above)
Comment (anttimaki) [reply apps/sertra/components/mod-selector/list.tsx]: Well that solves the issue, but I'd appreciate if you'd create the linting ticket anyway. Feel free to resolve this conversation once you've read this comment.

File: apps/sertra/components/mod-selector/search.tsx (hunk 2 above)
Comment (VilppeRiskidev) [reply apps/sertra/components/mod-selector/search.tsx]: Now using refs instead.

File: apps/sertra/components/mod-selector/search.tsx [hunk 5]
```diff
@@ -18,21 +18,15 @@ export const SearchBox = <T,>({
...
   const [isFocused, setFocused] = useState<boolean>(false);
 
-  function focusFirstOption(target: EventTarget & HTMLInputElement) {
-    if (
-      target.parentElement?.parentElement?.nextElementSibling?.children[0]
-        ?.children[0]
-    ) {
-      target.parentElement.parentElement.nextElementSibling.children[0].children[0].focus();
+  function focusFirstOption() {
+    if (optionsRef.current?.children[0]) {
```
Comment (anttimaki) [inline apps/sertra/components/mod-selector/search.tsx:22]: While the fix for useRef is implemented, you've seemed to ignore the comment about optional chaining operator. I guess we don't need to use it here, but please comment if the "wontfix" was intentional.

File: apps/sertra/components/mod-selector/search.tsx [hunk 6]
```diff
@@ -105,7 +103,7 @@ export const SearchBox = <T,>({
       </div>
       {isFocused && (
         <div className={styles.optionsAnchor}>
-          <div className={styles.optionsBox}>
+          <div className={styles.optionsBox} ref={optionsRef}>
```
Comment (anttimaki) [inline apps/sertra/components/mod-selector/search.tsx:106]: I guess this approach is fine, but I'm wondering if we could bind the ref directly to the element were interested in with something like this:

```
{options.map((option, i) => (
  <button
    ref={i === 0 ? optionRef : null}
    className={styles.optionButton}
    key={keyExtractor(option)}
    onKeyDown={(e) => listKeyDownHandler(option, e)}
    onClick={(e) => onClickHandler(option, e.currentTarget)}
  >
    {renderOption(option)}
  </button>
)}
```

Haven't tested, might not work 🤷🏻

**Review by anttimaki — APPROVED** [review]
Approved as is, use your own judgement regarding the new comments.

File: apps/sertra/components/mod-selector/list.tsx (hunk 1 above)
Comment (MythicManiac) [reply apps/sertra/components/mod-selector/list.tsx]: I know my IDEs at least auto-insert the braces regardless of value type, so as with any other formatting rules it'd be best if whatever formatting is linted for is also automated. In other words, a lint rule alone is not a sufficient change, the formatting should also be handled by prettier. If both can't be done then I don't think we should have a lint rule.
